src/viewport.js

```diff
@@ -1554,6 +1550,9 @@ $.Viewport.prototype = {
      */
     setFlip: function( state ) {
       if ( this.flipped != state ) {
+        this.flipped = state;
```

Comment:
You can just start this function with: 

```
if (this.flipped === state) {
    return;
}
```

That should help clean the logic up here.